File: scripts/test_solutions/133_clone_graph.py
def cloneGraph(node):
    if node is None:
        return None
    visited = {}

    def dfs(n):
        if n.val in visited:
            return visited[n.val]
        clone = Node(n.val)
        visited[n.val] = clone
        for neighbor in sorted(n.neighbors, key=lambda x: x.val):
            clone.neighbors.append(dfs(neighbor))
        return clone

    return dfs(node)

File: scripts/test_solutions/133_clone_graph.js
var cloneGraph = function(node) {
    if (!node) return null;
    const map = new Map();
    function dfs(n) {
        if (map.has(n.val)) return map.get(n.val);
        const clone = { val: n.val, neighbors: [] };
        map.set(n.val, clone);
        for (const neighbor of n.neighbors) clone.neighbors.push(dfs(neighbor));
        return clone;
    }
    return dfs(node);
};
